library/Notifications/Widget/ItemList/ContactGroupList.php

```diff
@@ -14,4 +15,10 @@ protected function getItemClass(): string
     {
         return ContactGroupListItem::class;
     }
+
+    protected function init(): void
+    {
+        $this->getAttributes()
+            ->set('data-icinga-detail-url', (string) Url::fromPath('notifications/contact-group'));
```

Comment:
I would not reintroduce a static url path and instead use the already existing reference in `Icinga\Module\Notifications\Common\Links` at line `89`.
This minifies errors for the case where the route definition could get changed in the future. It also helps with maintainability.
You also need to allow null values for use-cases where only the url reference is needed without any parameters, just like in this case.

A modification of that function could look like the following:
```php
/* add the ? modifier to the type definition, check for the existence of the id param, return the url without the id in case of its absence */
public static function contactGroup(?int $id): Url
{
    if ($id === null) {
        return Url::fromPath('notifications/contact-group');
    }

    return Url::fromPath('notifications/contact-group', ['id' => $id]);
}
```